homeassistant/components/climate/ambiclimate.py

```diff
@@ -0,0 +1,271 @@
+STORAGE_KEY = 'ambiclimate_auth'
+STORAGE_VERSION = 1
+
+AUTH_CALLBACK_NAME = 'api:ambiclimate'
+AUTH_CALLBACK_PATH = '/api/ambiclimate'
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE |
+                 SUPPORT_ON_OFF)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CLIENT_ID): cv.string,
+    vol.Required(CONF_CLIENT_SECRET): cv.string,
+})
+
+SEND_COMFORT_FEEDBACK_SCHEMA = vol.Schema({
+    vol.Required(ATTR_NAME): cv.string,
+    vol.Required(ATTR_VALUE): cv.string,
+})
+
+SET_COMFORT_MODE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_NAME): cv.string,
+})
+
+SET_TEMPERATURE_MODE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_NAME): cv.string,
+    vol.Required(ATTR_VALUE): cv.string,
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the Ambiclimate heater."""
+    import ambiclimate
+    callback_url = '{}{}'.format(hass.config.api.base_url, AUTH_CALLBACK_PATH)
+    oauth = ambiclimate.AmbiclimateOAuth(config.get(CONF_CLIENT_ID),
+                                         config.get(CONF_CLIENT_SECRET),
+                                         callback_url,
+                                         async_get_clientsession(hass))
+
+    store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
+    token_info = await store.async_load()
+
+    if not token_info:
+        _LOGGER.info("no token; requesting authorization")
+        hass.http.register_view(AmbiclimateAuthCallbackView(config,
+                                                            async_add_entities,
+                                                            oauth,
+                                                            store))
+
+        def _call_request_config():
+            configurator = hass.components.configurator
+            req = configurator.request_config
+            data = req(DEFAULT_NAME, lambda _: None,
+                       link_name=CONFIGURATOR_LINK_NAME,
+                       link_url=oauth.get_authorize_url(),
+                       description=CONFIGURATOR_DESCRIPTION,
+                       submit_caption=CONFIGURATOR_SUBMIT_CAPTION)
+            hass.data[AMBICLIMATE_DATA] = data
+
+        hass.async_add_job(_call_request_config)
+        return
+
+    if hass.data.get(AMBICLIMATE_DATA):
+        def _call_request_config_done():
+            configurator = hass.components.configurator
+            configurator.request_done(hass.data.get(AMBICLIMATE_DATA))
+            del hass.data[AMBICLIMATE_DATA]
+
+        hass.async_add_job(_call_request_config_done)
+
+    websession = async_get_clientsession(hass)
+    data_connection = ambiclimate.AmbiclimateConnection(oauth,
+                                                        token_info=token_info,
+                                                        websession=websession)
+
+    if await data_connection.refresh_access_token():
```

Comment:
In looking at the library, this returns `False` if no refresh was needed or if it fails to refresh.  Does the integration break if it fails to refresh?